src/resource/factory/TextureFactory.cpp

```diff
@@ -8,10 +8,13 @@ std::shared_ptr<Resource> TextureFactory::ReadResource(uint32_t version, std::sh
     auto resource = std::make_shared<Texture>();
     std::shared_ptr<ResourceVersionFactory> factory = nullptr;
 
-    switch (version) {
-        case 0:
+    switch ((Version)version) {
+        case Version::Deckard:
             factory = std::make_shared<TextureFactoryV0>();
             break;
+        case Version::Spock:
```

Comment:
We should talk about whether this should be `Spock` or `Roy`.
In other words, are we going by this being the second version of _this_ resource, or it being introduced as part of SoH `Spock`. I'd probably prefer we go with `Roy` here.